src/main/java/dev/compactmods/machines/util/PlayerUtil.java

```diff
@@ -147,6 +150,38 @@ public static void teleportPlayerOutOfMachine(ServerLevel world, @Nonnull Server
         });
     }
 
+    public static void teleportPlayerBackToMachine(ServerLevel world, @Nonnull ServerPlayer serverPlayer) {
+        MinecraftServer serv = world.getServer();
+
+        final LazyOptional<IRoomHistory> history = serverPlayer.getCapability(Capabilities.ROOM_HISTORY);
+
+        if (!history.isPresent()) {
+            howDidYouGetThere(serverPlayer);
+            return;
+        }
+
+        history.ifPresent(hist -> {
+            if (hist.hasHistory()) {
+                final IRoomHistoryItem currentArea = hist.peek();
+                final Level machineLevel = currentArea.getMachine().level(serv);
+                final BlockPos machinePos = currentArea.getMachine().getBlockPosition();
+
+                if(machineLevel.getBlockEntity(machinePos) instanceof CompactMachineBlockEntity tile) {
+                    final var targetRoom = tile.getConnectedRoom();
+                    targetRoom.ifPresent(room -> {
+                        // No need to change history since we just warping player back!
+
+                        try {
+                            teleportPlayerIntoRoom(serv, serverPlayer, room, false);
+                        } catch (MissingDimensionException | NonexistentRoomException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                }
+            }
+        });
+    }
+
```

Comment:
Evaluating if this is necessary; if the config option isn't an option list between tele back and tele out, this should probably just be removed. If kept, the non-existent history conditional block should fall into returning the player back to their spawn (`teleportPlayerToRespawnOrOverworld`)